skimage/restoration/_nl_means_denoising.pyx

```diff
@@ -365,6 +366,53 @@ cdef inline double _integral_to_distance_3d(double[:, :, ::] integral,
     return max(distance, 0.0) / (s_cube_h_square)
 
 
+cdef inline double _integral_to_distance_4d(double [:, :, :, ::] integral,
+                                            Py_ssize_t time, Py_ssize_t pln,
+                                            Py_ssize_t row, Py_ssize_t col,
+                                            Py_ssize_t offset,
+                                            double s4_h_square) nogil:
+    """
+    References
+    ----------
+    J. Darbon, A. Cunha, T.F. Chan, S. Osher, and G.J. Jensen, Fast
+    nonlocal filtering applied to electron cryomicroscopy, in 5th IEEE
+    International Symposium on Biomedical Imaging: From Nano to Macro,
+    2008, pp. 1331-1334.
+
+    Jacques Froment. Parameter-Free Fast Pixelwise Non-Local Means
+    Denoising. Image Processing On Line, 2014, vol. 4, p. 300-326.
+
+    Used in _fast_nl_means_denoising_4d
+
+    The coefficients for the terms were determined using Eq. 54 of [1]_ as
+    implemented in the integral_image_coefficients function.
+
+    E. Tapia.  A note on the computation of high-dimensional integral images.
```

Comment:
```suggestion
    .. [1] E. Tapia. A note on the computation of high-dimensional integral images.
```
?

Reply:
I have now fixed these across all functions where they are present in d1f0b28